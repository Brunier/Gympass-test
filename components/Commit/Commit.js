import React from 'react';
import styled from "styled-components";
import Text from "../Text/Text";
import TextTitle from "../TextTitle/TextTitle";
import TextParagraph from "../TextParagraph/TextParagraph";
import moment from "moment/moment";

const Container = styled.div`
  padding: 10px;
  border: 2px solid #00757e;
  height: 200px;
  margin-bottom: 30px;
    
  a {
    text-decoration: none;
    color: inherit;
  }
  
 `;

export class Commit extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container>
                <div>
                    <TextTitle maxWidth={"100%"} fontSize={18}><a href={this.props.commit.url} target="_blank">{this.props.commit.sha}</a></TextTitle>
                    <TextParagraph maxHeight={120} height={120}>
                        {this.props.commit.commit.message}
                    </TextParagraph>
                </div>
                <div>
                    <Text fontSize={16} bold={true}>{this.props.commit.commit.author.name}</Text>
                    <Text fontSize={16} bold={true}> - </Text>
                    <Text fontSize={16} bold={true}>{moment(this.props.commit.commit.author.date).format('llll')}</Text>
                </div>
            </Container>
        )
    }
}

export default Commit;
