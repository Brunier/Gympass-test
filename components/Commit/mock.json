{
  "sha": "942a765b53662936679b8119ed4533d972298d2a",
  "commit": {
    "author": {
      "name": "Pedro Bullo Junior",
      "email": "pedro.bj@hotmail.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "message": "Workaround empty child",
    "tree": {
      "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
      "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
    },
    "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
    "comment_count": 0,
    "verification": {
      "verified": true,
      "reason": "valid"
    }
  },
  "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
}
