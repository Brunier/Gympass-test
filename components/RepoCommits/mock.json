{
  "data" : [
    {
      "sha": "942a765b53662936679b8119ed4533d972298d2a",
      "commit": {
        "author": {
          "name": "Pedro Bullo Junior",
          "email": "pedro.bj@hotmail.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "message": "Workaround empty child",
        "tree": {
          "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
    },
    {
      "sha": "c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
      "commit": {
        "author": {
          "name": "matc4",
          "email": "matias.s.cortes@gmail.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "message": "Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added",
        "tree": {
          "sha": "97dae56ead52ab59e12b98848f4e38d89265040f",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/97dae56ead52ab59e12b98848f4e38d89265040f"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
    }
  ]
}
