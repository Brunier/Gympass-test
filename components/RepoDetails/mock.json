{
  "name": "interview-test",
  "description": "Interview test project",
  "url": "https://api.github.com/repos/Gympass/interview-test",
  "commits_url": "https://api.github.com/repos/Gympass/interview-test/commits",
  "created_at": "2018-05-22T19:57:37Z",
  "updated_at": "2019-01-24T02:51:23Z",
  "language": "Javascript",
  "stars": 3,
  "watchers": 3,
  "forks": 4
}
