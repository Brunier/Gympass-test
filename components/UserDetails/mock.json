{
  "repos": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ],
  "name": "gympass",
  "img": "https://avatars3.githubusercontent.com/u/26311162?v=4",
  "loading": false,
  "notFound": false
}
