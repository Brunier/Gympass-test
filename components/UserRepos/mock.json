{
  "data": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "devise",
      "description": "Flexible authentication solution for Rails with Warden.",
      "url": "https://api.github.com/repos/Gympass/devise",
      "commits_url": "https://api.github.com/repos/Gympass/devise/commits",
      "created_at": "2018-01-17T00:17:46Z",
      "updated_at": "2018-01-17T00:17:52Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "AvaTax-REST-V2-Ruby-SDK",
      "description": "Ruby SDK for AvaTax REST v2",
      "url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK",
      "commits_url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK/commits",
      "created_at": "2018-01-09T23:23:04Z",
      "updated_at": "2018-01-09T23:23:06Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ]
}
