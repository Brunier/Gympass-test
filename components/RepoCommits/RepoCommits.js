import React from 'react';
import Commit from "../Commit/Commit"


class RepoCommits extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <div>
                {this.props.commits.map((commit, i) => {
                    return (
                        <Commit key={commit.sha + i } commit={commit} />
                    )
                })}
            </div>
        )
    }
}

export default RepoCommits;
