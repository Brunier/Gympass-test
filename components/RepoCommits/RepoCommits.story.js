import React from 'react';
import { storiesOf } from '@storybook/react';
import RepoCommits from './RepoCommits';
import mock from "./mock"


storiesOf('RepoCommits', module)
    .add('Normal', () => (
        <RepoCommits  commits={mock.data}/>
    ))
