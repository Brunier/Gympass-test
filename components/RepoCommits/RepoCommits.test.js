import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import mock from "./mock"
// Components
import RepoCommits from './RepoCommits';
import Commit from '../Commit/Commit';

function setup(props) {
    const wrapper = shallow(<RepoCommits {...props} />);
    return { wrapper };
}

const commits = mock.data;


describe('RepoCommits Test Suite', () => {
    it('Verificar a quantidade de RepoCommits', () => {
        const { wrapper } = setup({commits});
        expect(wrapper.find(Commit)).toHaveLength(commits.length);
    });

    it('Snapshot', () => {
        const tree = renderer.create(<RepoCommits commits={commits} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
