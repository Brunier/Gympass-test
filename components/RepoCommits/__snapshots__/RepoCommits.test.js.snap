// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCommits Test Suite Snapshot 1`] = `
<div>
  <div
    className="sc-bxivhb bqVlmP"
  >
    <div>
      <h1
        className="sc-bwzfXH bJRMPj"
        fontSize={18}
      >
        <a
          href="https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
          target="_blank"
        >
          942a765b53662936679b8119ed4533d972298d2a
        </a>
      </h1>
      <p
        className="sc-htpNat jewUiJ"
        height={120}
      >
        Workaround empty child
      </p>
    </div>
    <div>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
        Pedro Bullo Junior
      </span>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
         - 
      </span>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
        Mon, Jan 21, 2019 8:59 PM
      </span>
    </div>
  </div>
  <div
    className="sc-bxivhb bqVlmP"
  >
    <div>
      <h1
        className="sc-bwzfXH bJRMPj"
        fontSize={18}
      >
        <a
          href="https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
          target="_blank"
        >
          c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c
        </a>
      </h1>
      <p
        className="sc-htpNat jewUiJ"
        height={120}
      >
        Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added
      </p>
    </div>
    <div>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
        matc4
      </span>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
         - 
      </span>
      <span
        className="sc-bdVaJa iDsqdx"
        fontSize={16}
      >
        Thu, Sep 13, 2018 9:34 AM
      </span>
    </div>
  </div>
</div>
`;
